sale_backorder/models/sale.py

```diff
@@ -0,0 +1,154 @@
+
+    @api.multi
+    @api.depends('order_line.uigd_qty', 'order_line.price_unit')
+    def _compute_uigd_value(self):
+        for order in self:
+            total = 0
+            for line in order.order_line:
+                total += line.uigd_value
+            order.uigd_value = total
+
+    @api.multi
+    @api.depends('order_line.bo_qty', 'order_line.price_unit')
+    def _compute_bo_value(self):
+        for order in self:
+            total = 0
+            for line in order.order_line:
+                total += line.bo_value
+            order.bo_value = total
+
+    @api.multi
+    @api.depends('order_line.last_date_delivered', 'order_line.qty_delivered',
+                 'order_line.qty_invoiced')
+    def _compute_last_date_delivered(self):
+        for order in self:
+            max_date = False
+            for line in order.order_line:
+                if max_date:
+                    if line.last_date_delivered and \
+                            max_date < line.last_date_delivered:
+                        max_date = line.last_date_delivered
+                else:
+                    max_date = line.last_date_delivered
+            order.last_date_delivered = max_date
+
+    @api.depends('order_line.last_bill_date', 'order_line.qty_delivered',
+                 'order_line.qty_invoiced')
+    def _compute_last_bill_date(self):
+        for order in self:
+            max_date = False
+            for line in order.order_line:
+                if max_date:
+                    if line.last_bill_date and max_date < line.last_bill_date:
+                        max_date = line.last_bill_date
+                else:
+                    max_date = line.last_bill_date
+            order.last_bill_date = max_date
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    last_date_delivered = fields.Datetime(
+        string="Last Date Delivered",
+        compute="_compute_last_date_delivered", store=True)
+    last_bill_date = fields.Datetime(
+        string="Last Bill Date",
+        compute="_compute_last_bill_date", store=True)
+    uigd_qty = fields.Float(string="UIGD Qty",
+                            compute='_compute_uigd_qty', store=True)
+    bo_qty = fields.Float(string="Backorder Qty",
+                          compute='_compute_bo_qty', store=True)
+    uigd_value = fields.Monetary(string="UIGD Value",
+                                 compute='_compute_uigd_value', store=True)
+    bo_value = fields.Monetary(string="Backorder Value",
+                               compute='_compute_bo_value', store=True)
+    prod_type = fields.Selection(string="Product Type",
+                                 selection=PRODUCT_TYPE_SELECTION,
+                                 related="product_id.type")
+    product_type = fields.Selection(string='Product Type',
+                                    related='product_id.product_tmpl_id.type')
```

Comment:
@smangukiya Why do we need 2 fields here? The template and the product always have the same type.